modules/apps/commerce/commerce-product-ddm-impl/src/main/java/com/liferay/commerce/product/ddm/internal/DDMHelperImpl.java

```diff
@@ -235,51 +234,50 @@ private String _createDDMFormRuleInputMapping(
 
 		List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
 
-		Stream<DDMFormField> stream = ddmFormFields.stream();
-
-		Stream<String> inputMappingStatementStream = stream.map(
-			field -> String.format(
-				"'%s=', getValue('%s')", field.getName(), field.getName()));
-
-		inputMappingStatementStream = Stream.concat(
```

Comment:
@julschong Just noticed one thing: the original logic always concat the previous stream to the end of the new stream, so the final output should be in reversed order.

However, although you put the for-loop to the end, you didn't reverse the `*id` parts.

To make this change, we'd better add a unit test to ensure the output before/after.

Reply:
Or if this is too hard to read, I'm also OK if you use `StringUtil.replace` and `StringUtil.merge`, at least they are still much lighter than `String.format`.